fdbclient/ManagementAPI.actor.cpp

```diff
@@ -280,6 +280,7 @@ ACTOR Future<DatabaseConfiguration> getDatabaseConfiguration( Database cx ) {
 ACTOR Future<ConfigurationResult::Type> changeConfig( Database cx, std::map<std::string, std::string> m, bool force ) {
 	state StringRef initIdKey = LiteralStringRef( "\xff/init_id" );
 	state Transaction tr(cx);
+	cx->setOption(FDBDatabaseOptions::USE_PROVISIONAL_PROXIES, Optional<StringRef>());
```

Comment:
This seems like an unusual pattern for setting database options. I think I'm ok with the idea of this being a database option, although based on its usages here it really seems like it would be more natural as a transaction option.

What is the reason for preferring database to transaction options?

Reply:
It has to be a database option, because the result of this option is a change to the list of masterProxies. The database object could keep two separate lists of proxies, but that seems like unnecessary overhead

Reply:
Maybe I'm misunderstanding, but it seems like this is just returning am empty list when you aren't using provisional proxies. If that's the case, you wouldn't need two lists but just the ability to detect if the list is provisional.

Then getMasterProxies could take useProvisionalProxies as an argument.

Reply:
From your comment, I started going down this road, but things get pretty ugly with read version batching. I would need to create separate batchers based on this option, this easiest would be to mirror this option as a ReadVersionOption, and verify that the proxy ignores will ignore flag bytes that are not relevant.

Reply:
Okay, I think I am okay with duplicating the option in the read version options, I am pretty sure it isn't going to cause any problems

Reply:
I hadn't considered the read version batching. It's a little unfortunate because we could end up with a combinatorial number of batchers based on the possible flag choices (currently 12 now, I believe), but I think it's probably ok given the intended rarity of this option. If it became necessary, it seems possible to rework the batching mechanism to not require multiple batchers for this at the cost of making it more complex, so I don't think we need to worry about it too much.